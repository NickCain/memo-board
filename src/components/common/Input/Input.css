.input {
    background: transparent;
    border: none;
    height: 25px;
}

.input:focus {
    outline: 2px solid #d3d3d3;
}
