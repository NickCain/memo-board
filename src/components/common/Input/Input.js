import PropTypes from 'prop-types';
import React, { Component } from 'react';

import './Input.css';

class Input extends Component {
  render() {
    const { placeholder, onBlur, onKeyPress, autoFocus, text } = this.props;
    return (
      <input
        className="input"
        autoFocus={autoFocus}
        onBlur={onBlur}
        onKeyPress={onKeyPress}
        placeholder={placeholder}
        defaultValue={text}
      />
    );
  }
}

Input.propTypes = {
  text: PropTypes.string,
  placeholder: PropTypes.string,
  autoFocus: PropTypes.bool,
  onBlur: PropTypes.func,
  onKeyPress: PropTypes.func
};

export default Input;
