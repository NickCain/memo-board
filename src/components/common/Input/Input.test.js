import { mount, shallow } from 'enzyme';
import React from 'react';
import Input from './Input.js'

let wrapper;
let shallowWrapper;

describe('Input - component', () => {
  describe('Input renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <Input />
      );
      shallowWrapper = shallow(
        <Input />
      );
    });

    it('should render the Input component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
