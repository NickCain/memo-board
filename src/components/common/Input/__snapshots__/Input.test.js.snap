// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input - component Input renders should render the Input component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    autoFocus={undefined}
    className="input"
    defaultValue={undefined}
    onBlur={undefined}
    onKeyPress={undefined}
    placeholder={undefined}
/>,
  "nodes": Array [
    <input
      autoFocus={undefined}
      className="input"
      defaultValue={undefined}
      onBlur={undefined}
      onKeyPress={undefined}
      placeholder={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Input />,
      "_debugID": 4,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Input {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          autoFocus={undefined}
          className="input"
          defaultValue={undefined}
          onBlur={undefined}
          onKeyPress={undefined}
          placeholder={undefined}
/>,
        "_debugID": 5,
        "_renderedOutput": <input
          autoFocus={undefined}
          className="input"
          defaultValue={undefined}
          onBlur={undefined}
          onKeyPress={undefined}
          placeholder={undefined}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Input />,
}
`;
